scio-smb/src/main/java/org/apache/beam/sdk/extensions/smb/SortedBucketSink.java

```diff
@@ -651,15 +647,36 @@ static void moveFiles(
       }
 
       LOG.info("Moving {} bucket files into {}", srcFiles.size(), dstFileAssignment.getDirectory());
-      try {
-        FileSystems.rename(srcFiles, dstFiles);
-      } catch (IOException e) {
-        cleanupTempFiles(e, filesToCleanUp);
-        throw new RuntimeException("Failed to rename temporary files", e);
-      }
-
+      // During a failure case, files may have been deleted in an earlier step. Thus
+      // we ignore missing files here.
+      FileSystems.rename(
+          srcFiles,
+          dstFiles,
+          StandardMoveOptions.IGNORE_MISSING_FILES,
```

Comment:
so would an SMB write ever get into a state where some bucket filse are missing in the final output directory?

Reply:
Yeah it's possible If the job is killed after 4 failures of this DoFn. But in that case the job will fail and output shouldn't be pushed to our data catalog.

Also this is assuming GCS rename of single files is atomic, which I believe is the case. So we won't end up with truncated files due to failure.